freud/common.pyx

```diff
@@ -125,6 +133,70 @@ cdef class Compute:
         return wrapper
 
 
+cdef class PairCompute(Compute):
+    R"""Parent class for all compute classes in freud that depend on finding
+    nearest neighbors.
+
+    The purpose of this class is to consolidate some of the logic for parsing
+    the numerous possible inputs to the compute calls of such classes. In
+    particular, this class contains a helper function that calls the necessary
+    functions to create NeighborQuery and NeighborList classes as needed, as
+    well as dealing with boxes and query arguments.
+
+    .. moduleauthor:: Vyas Ramasubramani <vramasub@umich.edu>
+    """
+    def preprocess_arguments(self, box, points, query_points=None, nlist=None,
+                              query_args=None):
+        """Process standard compute arguments into freud's internal types by
+        calling all the required internal functions.
+
+        The preprocessing of boxes and points into NeighborQuery objects, the
+        determination of how to handle the NeighborList object, the creation
+        of default query arguments as needed, deciding what query_points are,
+        and setting the appropriate exclude_ii flag are all handled in this
+        function.
+
+        Args:
+            box (:class:`freud.box.Box`):
+                Simulation box.
+            points ((:math:`N_{points}`, 3) :class:`numpy.ndarray`):
+                Reference points used to calculate the RDF.
+            query_points ((:math:`N_{query_points}`, 3) :class:`numpy.ndarray`, optional):
+                Points used to calculate the RDF. Uses :code:`points` if
+                not provided or :code:`None`.
+            nlist (:class:`freud.locality.NeighborList`, optional):
+                NeighborList to use to find bonds (Default value =
+                :code:`None`).
+            query_args (dict): A dictionary of query arguments (Default value =
+                :code:`None`).
+        """
+        cdef freud.box.Box b = freud.common.convert_box(box)
+
+        cdef freud.locality.NeighborQuery nq = freud.locality.make_default_nq(
+            box, points)
```

Comment:
This doesn't implement the choice of `NeighborQuery` or tuple syntax `(box, points)` yet, but we'll do that in this function when we get there, right?

Reply:
Correct, that will come in a future PR.